docs/how-to-guides/config-settings.rst

```diff
@@ -59,3 +59,63 @@ immediately install it, replace ``wheel`` with ``install``.  See the
 
 .. _build: https://pypa-build.readthedocs.io/en/stable/
 .. _pip: https://pip.pypa.io/
+
+
+Using a persistent build directory
+==================================
+
+By default, ``meson-python`` will create a temporary build directory, which
+gets cleaned up after the sdist or wheel has been built, or if the build fails.
```

Comment:
I would tweak this to "By default, meson-python creates a temporary build directory, which is deleted when the build terminates." It is a bit more direct and avoid the expression "cleaned up" which is not well defined.